contracts/genie-raffle/src/contract.rs

```diff
@@ -0,0 +1,391 @@
+    }
+    if cw20_msg.amount.is_zero() {
+        return Err(StdError::generic_err("amount must be greater than 0"));
+    }
+
+    match from_binary(&cw20_msg.msg)? {
+        Cw20HookMsg::IncreaseIncentives {} => {
+            handle_increase_cw20_incentives(deps, env, config.asset.asset_string(), cw20_msg.amount)
+        }
+    }
+}
+
+pub fn handle_increase_cw20_incentives(
+    deps: DepsMut,
+    env: Env,
+    asset: String,
+    amount: Uint128,
+) -> Result<Response, StdError> {
+    if query_status(deps.as_ref(), &env)?.status != Status::NotStarted {
+        return Err(StdError::generic_err(
+            "rewards can only be deposited before campaign starts",
+        ));
+    }
+    let mut state = STATE.load(deps.storage)?;
+    state.protocol_funding += amount;
+    STATE.save(deps.storage, &state)?;
+
+    Ok(Response::new().add_attributes(vec![
+        attr("action", "genie_increase_rewards"),
+        attr("asset", asset),
+        attr("protocol_funding", state.protocol_funding),
+    ]))
+}
+
+pub fn handle_increase_native_incentives(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+) -> Result<Response, StdError> {
+    if query_status(deps.as_ref(), &env)?.status != Status::NotStarted {
+        return Err(StdError::generic_err(
+            "rewards can only be deposited before campaign starts",
+        ));
+    }
+
+    let config = CONFIG.load(deps.storage)?;
+    if info.sender != config.owner {
+        return Err(StdError::generic_err("can only be called by owner"));
+    }
+
+    let increase_amount: Uint128 = match config.asset.clone() {
+        AssetInfo::NativeToken { denom } => info
+            .funds
+            .iter()
+            .filter(|coin| coin.denom == denom)
+            .map(|coin| coin.amount)
+            .sum(),
+        AssetInfo::Token { contract_addr: _ } => {
+            return Err(StdError::generic_err("invalid asset type"));
+        }
+    };
+    if increase_amount.is_zero() {
+        return Err(StdError::generic_err("amount must be greater than 0"));
+    }
+
+    let mut state = STATE.load(deps.storage)?;
+    state.protocol_funding += increase_amount;
+    STATE.save(deps.storage, &state)?;
+
+    Ok(Response::new().add_attributes(vec![
+        attr("action", "genie_increase_rewards"),
+        attr("asset", config.asset.asset_string()),
+        attr("protocol_funding", state.protocol_funding),
+    ]))
+}
+
+pub fn handle_claim(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    claim_amounts: Binary,
+    signature: Binary,
+) -> Result<Response, StdError> {
+    if query_status(deps.as_ref(), &env)?.status != Status::Ongoing {
+        return Err(StdError::generic_err("campaign is not ongoing"));
+    }
+
+    // convert claim_amounts to string
+    let claim_string = String::from_utf8(claim_amounts.to_vec())?;
+    let claim_string_array = claim_string.split(",").collect::<Vec<&str>>();
```

Comment:
nit: could `split` and `map` to ints in the same line instead of collecting to this vector of strings